daemon/api_snapshots_test.go

```diff
@@ -34,38 +34,21 @@ import (
 
 	"github.com/snapcore/snapd/client"
 	"github.com/snapcore/snapd/daemon"
-	"github.com/snapcore/snapd/dirs"
-	"github.com/snapcore/snapd/overlord"
-	"github.com/snapcore/snapd/overlord/assertstate"
 	"github.com/snapcore/snapd/overlord/snapshotstate"
-	"github.com/snapcore/snapd/overlord/snapstate"
 	"github.com/snapcore/snapd/overlord/state"
-	"github.com/snapcore/snapd/store/storetest"
 )
 
 var _ = check.Suite(&snapshotSuite{})
 
 type snapshotSuite struct {
-	d *daemon.Daemon
-	o *overlord.Overlord
+	apiBaseSuite
 }
 
 func (s *snapshotSuite) SetUpTest(c *check.C) {
-	s.o = overlord.Mock()
-	s.d = daemon.NewWithOverlord(s.o)
-
-	st := s.o.State()
-	// adds an assertion db
-	assertstate.Manager(st, s.o.TaskRunner())
-	st.Lock()
-	defer st.Unlock()
-	snapstate.ReplaceStore(st, storetest.Store{})
-	dirs.SetRootDir(c.MkDir())
-}
-
-func (s *snapshotSuite) TearDownTest(c *check.C) {
-	s.o = nil
-	s.d = nil
+	s.apiBaseSuite.SetUpTest(c)
+	d := s.daemonWithStore(c, s)
```

Comment:
looking at the old code it seems we might need a slightly different variant of this that takes an Overlord instead? I haven't needed it yet, but doesn't mean we can't add it